cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/AggregatedMetricsCollectionService.java

```diff
@@ -86,7 +91,33 @@ protected final void runOneIteration() throws Exception {
     Iterator<MetricValue> metricsItor = getMetrics(timestamp);
 
     try {
-      publish(metricsItor);
+      publish(metricsItor, new MetaMetricsComputer() {
+        private List<MetricValue> metaMetrics = Lists.newArrayList();
+        private int numMetrics = 0;
+
+        @Override
+        public void visitMetric(MetricValue metricValue) {
+          numMetrics++;
+
+          long timestampMs = TimeUnit.MILLISECONDS.convert(metricValue.getTimestamp(), TimeUnit.SECONDS);
+          long nowMs = System.currentTimeMillis();
+          long delayMs = nowMs - timestampMs;
+
+          MetricValue processDelayMetric = new MetricValue(
+            ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, "system"),
+            "metrics.global.processed.delay.ms", timestamp, delayMs, MetricType.GAUGE);
+          metaMetrics.add(processDelayMetric);
```

Comment:
Do you mean we may be storing too many `metaMetrics` in memory? Maybe we can have meta metrics per metric and per metric batch then?